vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -28,6 +28,29 @@
 logger = logging.getLogger(__name__)
 
 
+def _is_hidden(components: List[html.Div]):
+    """Returns True if all components are either None and/or have hidden=True."""
+    return all(div is None or getattr(div, "hidden", False) for div in components)
+
+
+# This is just used for type checking. Ideally it would inherit from some dash.development.base_component.Component
+# (e.g. html.Div) as well as TypedDict, but that's not possible, and Dash does not have typing support anyway. When
+# this type is used, the object is actually still a dash.development.base_component.Component, but this makes it easier
+# to see what contract the component fulfills by making the expected keys explicit.
+_PageDivsType = TypedDict(
```

Comment:
Notice the change for the alternative syntax of a TypedDict - it was required because we've decided to use hyphens instead of underscores for CSS classes and IDs. I'll create a separate PR where I'll consolidate it for the rest of the code basis, currently it's only implemented for the main page containers.